internal/artifacts/manager.go

```diff
@@ -10,23 +10,33 @@ import (
 	"os"
 	"path/filepath"
 	"sync"
+	"time"
 
 	"github.com/blang/semver/v4"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"go.uber.org/zap"
 	"golang.org/x/sync/singleflight"
 )
 
 // Manager supports loading, caching and serving Talos release artifacts.
 type Manager struct { //nolint:govet
-	options     Options
-	storagePath string
-	flavorsPath string
-	logger      *zap.Logger
+	options       Options
+	storagePath   string
+	flavorsPath   string
+	logger        *zap.Logger
+	imageRegistry name.Registry
+	pullers       map[Arch]*remote.Puller
 
-	flavorsSingleFlight singleflight.Group
+	sf singleflight.Group
```

Comment:
is this from new go?

Reply:
it's pretty old: https://pkg.go.dev/golang.org/x/sync/singleflight